src/tools/clippy/tests/ui/significant_drop_in_scrutinee.rs

```diff
@@ -314,6 +314,38 @@ fn should_trigger_lint_in_assign_expr() {
     };
 }
 
+#[derive(Debug)]
+enum RecursiveEnum {
+    Foo(Option<Box<RecursiveEnum>>)
+}
+
+#[derive(Debug)]
+enum GenericRecursiveEnum<T> {
+    Foo(T, Option<Box<GenericRecursiveEnum<T>>>)
+}
+
+fn should_not_cause_stack_overflow() {
+    /*let f = RecursiveEnum::Foo(Some(Box::new(RecursiveEnum::Foo(None))));
```

Comment:
Why is this commented out?

Reply:
That was an oversight on my part! Thank you!